packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -1,18 +1,124 @@
-import { createRequire } from "node:module";
+import assert from "node:assert";
+import { cloudflare } from "@cloudflare/unenv-preset";
 import MagicString from "magic-string";
 import { getNodeCompat } from "miniflare";
-import * as unenv from "unenv";
-import type { WorkerConfig } from "./plugin-config";
-import type { Environment } from "vite";
+import { defineEnv } from "unenv";
+import { resolvePluginConfig } from "./plugin-config";
+import type {
+	PluginConfig,
+	ResolvedPluginConfig,
+	WorkerConfig,
+} from "./plugin-config";
+import type { Plugin } from "vite";
+
+/**
+ * A Vite plugin that can provide Node.js compatibility support for Vite Environments that are hosted in Cloudflare Workers.
+ */
+export function nodejsCompatPlugin(pluginConfig: PluginConfig): Plugin {
+	let resolvedPluginConfig: ResolvedPluginConfig;
+	return {
+		name: "vite-plugin-nodejs-compat",
+		config(userConfig, env) {
+			// Capture the configuration of the Cloudflare Workers for use in other hooks, below.
+			resolvedPluginConfig = resolvePluginConfig(pluginConfig, userConfig, env);
```

Comment:
I don't think that we should be resolving this multiple times. Please can we move this plugin into `index.ts` and use the `resolvedPluginConfig` variable that is declared there.

Reply:
OK :sad:

Reply:
One reason I was thinking about this is that the plugin in this form is very nearly ready to be reused for the Vitest stuff.